scripts/task/KvToRdsMigration.php

```diff
@@ -0,0 +1,274 @@
+                    $deliveryExecutionIdentifier,
+                    $this->kvStorage->getTestTaker($deliveryExecutionIdentifier)
+                );
+            }
+            $this->deliveryExecutions[] = $deliveryExecutionIdentifier;
+        }
+
+        $this->report->add($this->migrateDeliveryExecutionVariables($callId, $deliveryExecutionIdentifier, $variables));
+    }
+
+    /**
+     * Migrate delivery execution variables
+     *
+     * If a variable already exists with same epoch then skip
+     * Otherwise register variable (item or test variable)
+     *
+     * @param $callId
+     * @param $deliveryExecutionIdentifier
+     * @param array $variables
+     * @return \common_report_Report
+     */
+    protected function migrateDeliveryExecutionVariables($callId, $deliveryExecutionIdentifier, array $variables)
+    {
+        $count = 0;
+        $identifier = null;
+
+        foreach ($variables as $variable) {
+            $identifier = $variable->variable->identifier;
+
+            foreach ($this->rdsStorage->getVariable($callId, $variable->variable->identifier) as $existingVariable) {
+                if ($variable->variable->epoch == $existingVariable->variable->epoch) {
+                    continue 2;
+                }
+            }
+            if (!$this->isDryrun()) {
+                if (isset($variable->callIdItem)) {
+                    $this->rdsStorage->storeItemVariable(
+                        $deliveryExecutionIdentifier,
+                        $variable->test,
+                        $variable->item,
+                        $variable->variable,
+                        $variable->callIdItem
+                    );
+                } else {
+                    $this->rdsStorage->storeTestVariable(
+                        $deliveryExecutionIdentifier,
+                        $variable->test,
+                        $variable->variable,
+                        $variable->callIdTest
+                    );
+                }
+            }
+            $count++;
+        }
+
+        if ($count == 0) {
+            $message = 'Already migrated.';
+        } else {
+            $message =  $count . ' variables migrated';
+        }
+
+        return \common_report_Report::createInfo('Migrating ' . $callId . ' : ' . $identifier . ' : ' . $message);
+    }
+
+    /**
+     * Get the configured service to deal with result storage
+     *
+     * @return \taoResultServer_models_classes_WritableResultStorage
+     * @throws \common_exception If the service is not a key value interface
+     */
+    protected function getCurrentKvResultStorage()
+    {
+        /** @var ResultServerService $resultService */
+        $resultService = $this->getServiceLocator()->get(ResultServerService::SERVICE_ID);
+        $resultStorageKey = $resultService->getOption(ResultServerService::OPTION_RESULT_STORAGE);
+        if ($resultStorageKey != 'taoAltResultStorage/KeyValueResultStorage') {
```

Comment:
I believe best practice is to always use strict checks as much as possible `!==`